src/MergeContext.js

```diff
@@ -292,28 +315,23 @@ class MergeContext {
         return true;
     }
 
-    // If approved, returns the number for milliseconds to wait for,
-    // or '0', meaning 'ready'. If not approved or disqualified returns null.
+    // calculates approval status properties (this._approvalResult)
     async _checkApproved() {
         const collaborators = await GH.getCollaborators();
         const pushCollaborators = collaborators.filter(c => c.permissions.push === true);
         const requestedReviewers = this._prRequestedReviewers();
+        this._approvalResult.reset();
 
         for (let collaborator of pushCollaborators) {
             if (requestedReviewers.includes(collaborator.login)) {
                 this._log("requested core reviewer: " + collaborator.login);
-                return null;
+                this._approvalResult.assign("waiting for requested reviews", "pending", null);
```

Comment:
I think we should split assign() into two or three methods so that the code intent is clear and so that the second method can check for a non-null delay (while the other two do not need to bother with a delay). For example, Approval::block(), Approval::grantAfterTimeout(), and Approval::grantNow().

BTW, I am not yet sure that zero timeout needs special handling -- I am just trying to go with what you have proposed.